lib/puma/request.rb

```diff
@@ -97,6 +97,8 @@ def handle_request(client, requests)
       env[RACK_AFTER_REPLY] ||= []
       env[RACK_RESPONSE_FINISHED] ||= []
 
+      env["puma.deferred_request"] = -> { worker.defer! }
```

Comment:
Defer is a weird verb here because we're not _deferring_ work to _later_. When someone calls this, they've either just performed or are about to perform an I/O call which is about to make this thread idle for an extended period of time. The name should reflect that.

Suggestions:

```
puma.sleepy_request
puma.park
puma.offload
puma.mark_io_bound
puma.move_to_io_pool
puma.mark_idle
```